python/private/python.bzl

```diff
@@ -329,79 +559,299 @@ def _get_bazel_version_specific_kwargs():
+python.toolchain(
+    is_default = True,
+    python_version = "3.11",
+)
+
+use_repo(python, my_python_name = "python_3_11")
+```
+
+Then the python interpreter will be available as `my_python_name`.
+:::
 """,
-            attrs = {
-                "configure_coverage_tool": attr.bool(
-                    mandatory = False,
-                    doc = "Whether or not to configure the default coverage tool for the toolchains.",
-                ),
-                "ignore_root_user_error": attr.bool(
-                    default = False,
-                    doc = """\
-If False, the Python runtime installation will be made read only. This improves
+    attrs = {
+        "configure_coverage_tool": attr.bool(
+            mandatory = False,
+            doc = "Whether or not to configure the default coverage tool provided by `rules_python` for the compatible toolchains.",
+        ),
+        "ignore_root_user_error": attr.bool(
+            default = False,
+            doc = """\
+If `False`, the Python runtime installation will be made read only. This improves
 the ability for Bazel to cache it, but prevents the interpreter from creating
-pyc files for the standard library dynamically at runtime as they are loaded.
+`.pyc` files for the standard library dynamically at runtime as they are loaded.
 
-If True, the Python runtime installation is read-write. This allows the
-interpreter to create pyc files for the standard library, but, because they are
+If `True`, the Python runtime installation is read-write. This allows the
+interpreter to create `.pyc` files for the standard library, but, because they are
 created as needed, it adversely affects Bazel's ability to cache the runtime and
 can result in spurious build failures.
 """,
-                    mandatory = False,
-                ),
-                "is_default": attr.bool(
-                    mandatory = False,
-                    doc = "Whether the toolchain is the default version",
-                ),
-                "python_version": attr.string(
-                    mandatory = True,
-                    doc = "The Python version, in `major.minor` format, e.g " +
-                          "'3.12', to create a toolchain for. Patch level " +
-                          "granularity (e.g. '3.12.1') is not supported.",
-                ),
-            },
+            mandatory = False,
+        ),
+        "is_default": attr.bool(
+            mandatory = False,
+            doc = "Whether the toolchain is the default version",
+        ),
+        "python_version": attr.string(
+            mandatory = True,
+            doc = """\
+The Python version, in `major.minor` or `major.minor.patch` format, e.g
+`3.12` (or `3.12.3`), to create a toolchain for.
+""",
         ),
     },
+)
+
+_override = tag_class(
+    doc = """Tag class used to override defaults and behaviour of the module extension.
+
+:::{versionadded} 0.36.0
+:::
+""",
+    attrs = dict(
```

Comment:
nit: use `{}` literal with `|`. It's just a bit less indentation and flows a bit nicer.
```
attrs = {
  "key": ...
} | OTHER_DICT
```